internal/metrics/update.go

```diff
@@ -168,13 +182,15 @@ func updateMetrics(metrics map[string]*metricData) {
 				for _, metricElement := range metricType.Elements {
 
 					// Clear existing metric values
-					metric := metrics[makeKey(metricElement)]
-					metric.values = make(map[string]float64)
-
-					// Update metric with cached values of publication data
-					for label, value := range metricElement.Values {
-						normalisedValue := mqmetric.Normalise(metricElement, label, value)
-						metric.values[label] = normalisedValue
+					metric, ok := metrics[makeKey(metricElement)]
+					if ok {
+						metric.values = make(map[string]float64)
+
+						// Update metric with cached values of publication data
+						for label, value := range metricElement.Values {
+							normalisedValue := mqmetric.Normalise(metricElement, label, value)
+							metric.values[label] = normalisedValue
+						}
 					}
```

Comment:
What about if we're not ok? Do we need an error. What does it mean?
Can we either have an else with error handling or comment as to why it doesn't matter.